src/js/Keyboard.js

```diff
@@ -29,12 +29,17 @@ class Keyboard extends Component {
   }
 
   handleAutoComplete = append => {
-    var completedWord = this.state.input + append;
+    var completedInput = ""
+    if (this.state.input[this.state.input.length-1] === " ") {
+      completedInput = this.state.input + append;
+    } else {
+      completedInput = this.state.input + " " + append;
+    }
```

Comment:
```suggestion
    var completedInput = append
```
This value should just replace the current input, not append